libs/blocks/merch/merch.js

```diff
@@ -24,86 +18,97 @@ const GEO_MAPPINGS = {
   no: 'nb-NO',
 };
 
-const PLACEHOLDER_TYPE_MAPPINGS = {
-  optical: 'priceOptical',
-  strikethrough: 'priceStrikethrough',
-};
-
-let initialized = false;
-
-function omitNullValues(target) {
+const omitNullValues = (target) => {
   if (target != null) {
     Object.entries(target).forEach(([key, value]) => {
       if (value == null) delete target[key];
     });
   }
   return target;
-}
+};
 
-const getTacocatEnv = (envName, locale) => {
-  const wcsLocale = GEO_MAPPINGS[locale.prefix] ?? locale.ietf;
-  // eslint-disable-next-line prefer-const
-  let [language, country = 'US'] = wcsLocale.split('-', 2);
+window.tacocat = window.tacocat || {};
+
+export const imsCountryPromise = () => new Promise((resolve) => {
+  let count = 0;
+  const check = setInterval(() => {
+    count += 1;
+    if (window.adobeIMS) {
+      clearInterval(check);
+      if (window.adobeIMS.isSignedInUser()) {
+        window.adobeIMS
+          .getProfile()
+          .then(({ countryCode }) => resolve(countryCode))
+          .catch(() => resolve());
+      } else {
+        resolve();
+      }
+    } else if (count > 25) {
+      clearInterval(check);
+      resolve();
+    }
+  }, 200);
+});
+window.tacocat.imsCountryPromise = imsCountryPromise();
+
+export const getTacocatEnv = (envName, locale) => {
+  const wcsLocale = (GEO_MAPPINGS[locale.prefix] ?? locale.ietf).split('-', 2);
+  let language = wcsLocale[0];
+  const country = wcsLocale[1] || 'US';
```

Comment:
can we do something like LOCALE_PATTERN=/(?<language>\w+)-(?<country>\w+)/ and match groups ? Talking of which, won't we have an issue with some languages that have - in their code?

Reply:
@npeltier this is simple enough.
also as per libs/scripts/scripts.js there is no language with `-` in it.

Reply:
it can happen that 'zh' will turn into 'zh-Hant' later, but we could guard against it just by splitting by the last found '-'. 
however it can be a future improvement for a next PR.

[https://github.com/adobecom/milo/blob/main/libs/scripts/scripts.js#L99](https://github.com/adobecom/milo/blob/main/libs/scripts/scripts.js#L99)

Reply:
yes, i tested this against the current scripts object.